src/index.js

```diff
@@ -363,7 +363,12 @@ function graphqlHTTP(options: Options): Middleware {
         // Note: Information about the error itself will still be contained in
         // the resulting JSON payload.
         // https://graphql.github.io/graphql-spec/#sec-Data
-        if (response.statusCode === 200 && result && !result.data) {
+        if (
+          response.statusCode === 200 &&
+          result &&
+          result.hasOwnProperty('errors') && // eslint-disable-line
+          !result.hasOwnProperty('data') // eslint-disable-line
```

Comment:
Follows GraphQL specs:

<img width="832" alt="Screenshot 2020-05-19 at 23 20 13" src="https://user-images.githubusercontent.com/10673347/82379505-4decfb80-9a27-11ea-9b65-0d787293e66c.png">

We only want to return 500 in the first case.

Reply:
Awesome! This solution seems pretty generic and is well justified by the specification. Are you going to submit a PR for the original repository? Simultaneously we can go ahead with our custom fork. To save work you could create a PR with the actual fix, commit it to this repository and then make another commit with the package.json updates

Reply:
> Are you going to submit a PR for the original repository?

I can definitely do, but they have PRs open since years and it really doesn't look active 🤷‍♂️ From what I read in the issues, the project was maintained by a Facebook guy that left.

> To save work you could create a PR with the actual fix, commit it to this repository and then make another commit with the package.json updates

Do you mean version change ? I was thinking about keeping the same version number than them so it's easier to trace back in the future. If so, this PR already changes what's needed to directly publish the package.

Reply:
ok, cool